src/main/java/org/cascadebot/cascadebot/data/objects/PermissionObject.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2020 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.data.objects;
+
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+
+/**
+ * Represents a data object that contains a per guild permission
+ */
+public interface PermissionObject {
+    String getPermission();
+
+    String getParent();
+
+    Module cascadeModule();
+
+    default CascadePermission getInternalPermission() {
+        return CascadePermission.of(getParent() + "." + getPermission(), false, cascadeModule());
+    }
+}
```

Comment:
I think we should start to make things like these abstract classes if they have method bodies. I believe we've talked about this before.